src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -121,20 +121,33 @@ public void removeCustomer(Customer key) {
      */
     public boolean hasCommission(Customer customer, Commission commission) {
         requireAllNonNull(customer, commission);
+        // Previously duplicate commission won't be detected because customer field has not been set.
+        commission.setCustomer(customer);
```

Comment:
Yeah, I agree this looks disgusting but if I don't set the customer here duplicate commission will not be detected because the equals method involves comparing the customer field as well so will cause nullpointerexception. when the duplicate is later detected in during the HashSet add. I did this because for now isSameCommission() only compares the title and not the customer. Else I can move the set customer to the addcommission in ModelManager which I think will prevent this issue where the selected customer is the owner of the commission. Just that it seems to break the abstraction layers?

Reply:
My main concern with this approach is the possibility that `hasCommission` changes the customer of commission wrongly if the commission belongs to a different customer, I think setting the customer within addcommission makes more sense since it is supposed to add the commission to the customer's commission list, and after it is added, the commission's customer should be the customer that it was just added to.